### PR TITLE
Gui: allow change ExpressionCompleter filter mode

### PR DESCRIPTION
**Pull request by realthunder**
Reported in https://forum.freecadweb.org/viewtopic.php?f=3&t=49953

### CODE REVIEW

File: src/Gui/ExpressionCompleter.cpp [hunk 1]
```diff
@@ -562,6 +570,14 @@ void ExpressionLineEdit::setNoProperty(bool enabled) {
         completer->setNoProperty(enabled);
 }
 
+void ExpressionLineEdit::setMatchExact(bool enabled) {
+    matchExact = enabled;
+#if QT_VERSION>=QT_VERSION_CHECK(5,2,0)
+    if (completer)
+        completer->setFilterMode(matchExact ? Qt::MatchExactly : Qt::MatchContains);
```
Comment (wwmayer) [inline src/Gui/ExpressionCompleter.cpp:577]: According to https://doc.qt.io/qt-5/qcompleter.html#filterMode-prop Qt::MatchExactly is not supported. It must be replaced with Qt::MatchStartsWith which is the default

File: src/Gui/Tree.cpp [hunk 2]
```diff
@@ -2833,6 +2833,7 @@ TreePanel::TreePanel(const char *name, QWidget* parent)
             this, SLOT(showEditor()));
 
     this->searchBox = new Gui::ExpressionLineEdit(this,true);
+    static_cast<ExpressionLineEdit*>(this->searchBox)->setMatchExact(false);
```
Comment (wwmayer) [inline src/Gui/Tree.cpp:2836]: This line doesn't make sense to me as it actively disallows to customize the search behaviour by setting the parameter value CompleterMatchExact

**Comment by wwmayer** [comment]
Fixed issues in 2c744264d and added possibility to change behaviour via context-menu

**Comment by realthunder** [comment]
Oops. Too quick with the document. Thanks for the fix.

**Pull request merged**